src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/NewAzureRmHubVirtualNetworkConnectionCommand.cs

```diff
@@ -0,0 +1,176 @@
+        DefaultParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+        SupportsShouldProcess = true),
+        OutputType(typeof(PSHubVirtualNetworkConnection))]
+    public class NewHubVirtualNetworkConnectionCommand : HubVnetConnectionBaseCmdlet
+    {
+        [Alias("ResourceName", "HubVirtualNetworkConnectionName")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource name.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Alias("VirtualHubName", "ParentVirtualHubName")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+            HelpMessage = "The parent resource name.")]
+        public string ParentResourceName { get; set; }
+
+        [Alias("VirtualHub", "ParentVirtualHub")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubObject,
+            HelpMessage = "The parent resource.")]
+        public PSVirtualHub ParentResource { get; set; }
+
+        [Alias("VirtualHubId", "ParentVirtualHubId")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubResourceId,
+            HelpMessage = "The parent resource id.")]
+        [ResourceIdCompleter("Microsoft.Network/virtualHubs")]
+        public string ParentResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The remote virtual network to which this hub virtual network connection is connected.")]
+        [ResourceGroupCompleter]
+        public PSVirtualNetwork RemoteVirtualNetwork { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The remote virtual network id to which this hub virtual network connection is connected.")]
+        [ResourceGroupCompleter]
+        [ResourceIdCompleter("Microsoft.Network/virtualNetworks")]
+        public string RemoteVirtualNetworkId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Enable internet security for this connection.")]
+        public bool? EnableInternetSecurity { get; set; }
```

Comment:
This should be a switchparameter, not a boolean.  Booleans should only be used on "Update" - switch parameter is correct for New and Set.  Please apply across the board.

Reply:
Fixed.

However, I have a question. How do we differetiate between
1. that user did not specify the paramter since it was not mandatory and he did not want to change the existing value.
AND
2. user did not specify the paramter because he wanted to change this value from true to false?


Is there a way to differentiate between the above two scenarios?